src/libcore/fmt/builders.rs

```diff
@@ -89,9 +89,9 @@ pub struct DebugStruct<'a, 'b: 'a> {
     has_fields: bool,
 }
 
-pub fn debug_struct_new<'a, 'b>(fmt: &'a mut fmt::Formatter<'b>,
-                                name: &str)
-                                -> DebugStruct<'a, 'b> {
+pub(crate) fn debug_struct_new<'a, 'b>(fmt: &'a mut fmt::Formatter<'b>,
```

Comment:
I'd avoid changing libcore until the `crate` sugar for `pub(crate)` is available in bootstrap compiler.